x-pack/test/saved_object_api_integration/common/lib/test_data_loader.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { FtrProviderContext } from '../ftr_provider_context';
+
+const SPACE_1 = {
+  id: 'space_1',
+  name: 'Space 1',
+  description: 'This is the first test space',
+  disabledFeatures: [],
+};
+
+const SPACE_2 = {
+  id: 'space_2',
+  name: 'Space 2',
+  description: 'This is the second test space',
+  disabledFeatures: [],
+};
+
+// Objects can only be imported in one space at a time. To have test saved objects
+// that are shared in multiple spaces we should import all objects in the "original"
+// spaces first and then share them to other spaces as a subsequent operation.
+const OBJECTS_TO_SHARE: Array<{
+  spacesToAdd?: string[];
+  spacesToRemove?: string[];
+  objects: Array<{ type: string; id: string }>;
+}> = [
+  {
+    spacesToAdd: ['*'],
+    spacesToRemove: ['default'],
+    objects: [
+      { type: 'sharedtype', id: 'all_spaces' },
+      { type: 'sharedtype', id: 'space_2_only_matching_origin' },
+      { type: 'sharedtype', id: 'alias_delete_exclusive' },
+    ],
+  },
+  {
+    spacesToRemove: ['default'],
+    spacesToAdd: [SPACE_1.id, SPACE_2.id],
+    objects: [{ type: 'sharedtype', id: 'space_1_and_space_2' }],
+  },
+  {
+    spacesToAdd: [SPACE_1.id, SPACE_2.id],
+    objects: [
+      { type: 'sharedtype', id: 'each_space' },
+      { type: 'sharedtype', id: 'conflict_2_all' },
+      { type: 'sharedtype', id: 'alias_delete_inclusive' },
+    ],
+  },
+  {
+    spacesToAdd: [SPACE_1.id],
+    objects: [
+      { type: 'sharedtype', id: 'conflict_1c_default_and_space_1' },
+      { type: 'sharedtype', id: 'default_and_space_1' },
+    ],
+  },
+  {
+    spacesToAdd: [SPACE_2.id],
+    objects: [{ type: 'sharedtype', id: 'default_and_space_2' }],
+  },
+];
+
+export function getTestDataLoader({ getService }: FtrProviderContext) {
```

Comment:
How identical is this file/function to [`x-pack/test/spaces_api_integration/common/lib/test_data_loader.ts`](https://github.com/elastic/kibana/blob/main/x-pack/test/spaces_api_integration/common/lib/test_data_loader.ts)?

Would it be possible to reuse it between tests?

Reply:
Great call! Done!